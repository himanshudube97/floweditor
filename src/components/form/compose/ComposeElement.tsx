import { react as bindCallbacks } from 'auto-bind';
import FormElement, { FormElementProps } from 'components/form/FormElement';
import * as React from 'react';
import { StringEntry } from 'store/nodeEditor';

export interface ComposeProps extends FormElementProps {
  name: string;
  entry?: StringEntry;
  errors?: string[];
  chatbox?: boolean;
  attachments?: boolean;
  counter?: boolean;
  onChange?: (composeValue: string) => void;
}

export default class ComposeElement extends React.Component<ComposeProps> {
  ele: HTMLElement;

  constructor(props: ComposeProps) {
    super(props);

    let initial = '';
    if (this.props.entry && this.props.entry.value) {
      initial = this.props.entry.value;
    }

    this.state = {
      value: initial
    };

    bindCallbacks(this, {
      include: [/^on/, /^handle/]
    });
  }

  public handleEventListenerRegistration(ref: any): void {
    if (!this.ele && ref) {
      this.ele = ref;
    }
  }

  public componentDidMount(): void {
    if (this.ele) {
      this.ele.addEventListener('temba-content-changed', this.handleChange);
    }
  }

  public handleChange(event: any): void {
    window.setTimeout(() => {
      if (this.props.onChange) {
        this.props.onChange(event.target.value);
      }
    }, 0);
  }

  public render(): JSX.Element {
    const optional: any = {};
    if (this.props.chatbox) {
      optional['chatbox'] = this.props.chatbox;
    }
    if (this.props.attachments) {
      optional['attachments'] = this.props.attachments;
    }
    if (this.props.counter) {
      optional['counter'] = this.props.counter;
    }

    return (
      <FormElement
        __className={this.props.__className}
        name={this.props.name}
        entry={this.props.entry}
      >
        <temba-compose
          ref={this.handleEventListenerRegistration}
          name={this.props.name}
          {...optional}
          value={this.props.entry.value}
          errors={this.props.errors}
          onChange={this.handleChange}
        ></temba-compose>
      </FormElement>
    );
  }
}
