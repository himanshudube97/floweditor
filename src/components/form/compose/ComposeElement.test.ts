import ComposeElement, { ComposeProps } from 'components/form/compose/ComposeElement';
import { composeComponentTestUtils } from 'testUtils';

const baseProps: ComposeProps = {
  name: 'Compose',
  entry: { value: '' }
};

const { setup } = composeComponentTestUtils(ComposeElement, baseProps);

const createWrapper = () => {
  return setup(false, {
    $merge: {
      chatbox: true,
      attachments: true,
      counter: true,
      // button: true,
      // handleButtonClicked: jest.fn(),
      onChange: jest.fn()
    }
  }).wrapper;
};

describe(ComposeElement.name, () => {
  it('renders', () => {
    const wrapper = createWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
