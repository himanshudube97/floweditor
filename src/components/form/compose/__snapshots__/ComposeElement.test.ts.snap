// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposeElement renders 1`] = `
<ComposeElement
  attachments={true}
  chatbox={true}
  counter={true}
  entry={
    Object {
      "value": "",
    }
  }
  name="Compose"
  onChange={[MockFunction]}
>
  <FormElement
    entry={
      Object {
        "value": "",
      }
    }
    name="Compose"
  >
    <div
      className="ele"
    >
      <temba-compose
        attachments={true}
        chatbox={true}
        counter={true}
        name="Compose"
        onChange={[Function]}
        value=""
      />
    </div>
  </FormElement>
</ComposeElement>
`;
